jwst/pipeline/calwebb_spec2.py

```diff
@@ -408,141 +433,199 @@ def process_exposure_product(
+        if not self.residual_fringe.skip and exp_type != "MIR_MRS":
+            self.log.debug(
+                "Science data does not allow residual fringe correction. "
+                'Skipping "residual fringe".'
+            )
             self.residual_fringe.skip = True
 
         # Apply the fringe correction for MIRI MRS
-        if not self.fringe.skip and exp_type != 'MIR_MRS':
+        if not self.fringe.skip and exp_type != "MIR_MRS":
             self.log.debug('Science data does not allow fringe correction. Skipping "fringe".')
             self.fringe.skip = True
 
         # Apply pathloss correction to MIRI LRS, NIRSpec, and NIRISS SOSS exposures
-        if not self.pathloss.skip and exp_type not in ['MIR_LRS-FIXEDSLIT', 'NRS_FIXEDSLIT',
-                                                       'NRS_MSASPEC', 'NRS_IFU', 'NIS_SOSS']:
+        if not self.pathloss.skip and exp_type not in [
+            "MIR_LRS-FIXEDSLIT",
+            "NRS_FIXEDSLIT",
+            "NRS_MSASPEC",
+            "NRS_IFU",
+            "NIS_SOSS",
+        ]:
             self.log.debug('Science data does not allow pathloss correction. Skipping "pathloss".')
             self.pathloss.skip = True
 
         # Apply barshadow correction to NIRSPEC MSA exposures
-        if not self.barshadow.skip and exp_type != 'NRS_MSASPEC':
-            self.log.debug('Science data does not allow barshadow correction. Skipping "barshadow".')
+        if not self.barshadow.skip and exp_type != "NRS_MSASPEC":
+            self.log.debug(
+                'Science data does not allow barshadow correction. Skipping "barshadow".'
+            )
             self.barshadow.skip = True
 
         # Apply master background only to NIRSPEC MSA exposures
-        if not self.master_background_mos.skip and exp_type != 'NRS_MSASPEC':
-            self.log.debug('Science data does not allow master background correction. Skipping "master_background_mos".')
+        if not self.master_background_mos.skip and exp_type != "NRS_MSASPEC":
+            self.log.debug(
+                "Science data does not allow master background correction. "
+                'Skipping "master_background_mos".'
+            )
             self.master_background_mos.skip = True
 
         # Apply WFSS contamination correction only to WFSS exposures
         if not self.wfss_contam.skip and exp_type not in WFSS_TYPES:
-            self.log.debug('Science data does not allow WFSS contamination correction. Skipping "wfss_contam".')
+            self.log.debug(
+                'Science data does not allow WFSS contamination correction. Skipping "wfss_contam".'
+            )
             self.wfss_contam.skip = True
 
     def _process_grism(self, data):
-        """WFSS & Grism processing
-
-        WFSS/Grism data need flat_field before extract_2d.
         """
+        Calibrate WFSS & Grism data.
```

Comment:
I recommend editing these three lines to read:

`Apply flat_field to calibrate WFSS and Grism data.`

Reply:
The intent of the "extended description" appears to be that it clarifies how grism processing differs from other modes.  I don't think the proposed edit retains that.  Instead, I've expanded the extended description to explicitly state "this differs from other modes because..."

let me know if it looks ok to you: https://github.com/spacetelescope/jwst/pull/9144/commits/74df406d695e58db4cbd8da1035e0f136168f886